### PR TITLE
Disable translation validation on performance runs.

### PR DESCRIPTION
**Pull request by ysiraichi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104887

This PR disables translation validation (TV) when running the benchmark suits on
performance workflows: inductor with A100s.

In summary, the changes are:

- Add flag for turning TV on and off on _benchmarks/dynamo/common.py_
- Turn TV on only on CI accuracy builds
- Add `--no-translation-validation` target flag to _.ci/pytorch/test.sh_

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78

### CODE REVIEW

**Comment by ysiraichi** [comment]
Perf run: https://github.com/pytorch/pytorch/actions/runs/5511272974

File: .ci/pytorch/test.sh [hunk 1]
```diff
@@ -348,6 +348,8 @@ test_perf_for_dashboard() {
       local target_flag=("--${target}")
       if [[ "$target" == "performance" ]]; then
         target_flag+=( --cold-start-latency)
+      elif [[ "$target" == "accuracy" ]]; then
+        target_flag+=( --no-translation-validation)
```
Comment (ezyang) [inline .ci/pytorch/test.sh:352]: we should do this for performance too

**Review by ezyang — APPROVED** [review]

File: .ci/pytorch/test.sh (hunk 1 above)
Comment (ysiraichi) [reply .ci/pytorch/test.sh:352]: But, we only turn TV on when we pass `--accuracy --ci` arguments. Should I still add the flag for performance?

File: benchmarks/dynamo/common.py [hunk 2]
```diff
@@ -2629,6 +2629,13 @@ def get_example_inputs(self):
         default=1,
         help="Set per-process GPU memory fraction (limit) for reducing usable size and reproducing OOMs",
     )
+
+    parser.add_argument(
+        "--no-translation-validation",
+        action="store_true",
+        help="Disable translation validation for accuracy builds.",
+    )
```
Comment (ysiraichi) [inline benchmarks/dynamo/common.py:2637]: Had to change this in order for it to work with Python 3.8, since, on that version, there's no `argparse.BooleanOptionalAction`.

**Comment by ysiraichi** [comment]
[dashboard link](https://hud.pytorch.org/benchmark/compilers?startTime=Tue%2C%2004%20Jul%202023%2012%3A12%3A10%20GMT&stopTime=Tue%2C%2011%20Jul%202023%2012%3A12%3A10%20GMT&granularity=hour&suite=torchbench&mode=training&dtype=amp&lBranch=gh/ysiraichi/62/orig&lCommit=4c92f9add4cd7aa5d08ba20e9f10b7affb152404&rBranch=main&rCommit=54f33265dba0d0782febe8bf8029efb2d0b29ac1)

I'm not really seeing any compilation time improvement, here. However, I'm pretty sure TV is not running anymore.

Before this PR, we can see in its logs that TV timeout-ed on accuracy runs, when training TIMM model `swin_base_patch4_window7_224` with dynamic shapes. That's not the case for this PR.

File: .ci/pytorch/test.sh (hunk 1 above)
Comment (ysiraichi) [reply .ci/pytorch/test.sh:352]: @ezyang Is it ok to leave it only for accuracy runs, since we don't turn TV on for performance runs?

Comment (ezyang) [reply .ci/pytorch/test.sh:352]: OK, fair enough.

**Comment by ezyang** [comment]
OK, SGTM
